backend/src/Manager/StoreOwnerBranch/StoreOwnerBranchManager.php

```diff
@@ -0,0 +1,112 @@
+
+    /**
+     * @param AutoMapping $autoMapping
+     * @param EntityManagerInterface $entityManager
+     * @param StoreOwnerBranchEntityRepository $storeOwnerBranchEntityRepository
+     * @param StoreOwnerProfileManager $storeOwnerProfileManager
+     */
+    public function __construct(AutoMapping $autoMapping, EntityManagerInterface $entityManager, StoreOwnerBranchEntityRepository $storeOwnerBranchEntityRepository, StoreOwnerProfileManager $storeOwnerProfileManager)
+    {
+        $this->autoMapping = $autoMapping;
+        $this->entityManager = $entityManager;
+        $this->storeOwnerBranchEntityRepository = $storeOwnerBranchEntityRepository;
+        $this->storeOwnerProfileManager = $storeOwnerProfileManager;
+    }
+
+    /**
+     * @param StoreOwnerBranchCreateRequest $request
+     * @return StoreOwnerBranchEntity|null
+     */
+    public function createBranch(StoreOwnerBranchCreateRequest $request):?StoreOwnerBranchEntity
+    {
+        $storeOwner = $this->storeOwnerProfileManager->getStoreOwnerProfile($request->getStoreOwner());
+     
+        $entity = $this->autoMapping->map(StoreOwnerBranchCreateRequest::class, StoreOwnerBranchEntity::class, $request);
+        $entity->setIsActive(1);
+        $entity->setStoreOwner($storeOwner);
+
+        $this->entityManager->persist($entity);
+        $this->entityManager->flush();
+        
+        return $entity;
+    }
+
+    /**
+     * @param StoreOwnerBranchUpdateRequest $request
+     * @return StoreOwnerBranchEntity|string
+     */
+    public function updateBranch(StoreOwnerBranchUpdateRequest $request): StoreOwnerBranchEntity|string
+    {
+        $entity = $this->storeOwnerBranchEntityRepository->find($request->getId());
+
+        if ($entity) {
+             
+            $entity = $this->autoMapping->mapToObject(StoreOwnerBranchUpdateRequest::class, StoreOwnerBranchEntity::class, $request, $entity);
+
+            $this->entityManager->flush();
+
+            return $entity;
+        }
+
+        return StoreOwnerBranch::BRANCH_NOT_FOUND;
+    }
+
+    /**
+     * @param StoreOwnerBranchDeleteRequest $request
+     * @return StoreOwnerBranchEntity|string
+     */
+    public function deletebranch(StoreOwnerBranchDeleteRequest $request): StoreOwnerBranchEntity|string
+    {
+        $entity = $this->storeOwnerBranchEntityRepository->find($request->getId());
+
+        if ($entity) {
+             
+            $entity = $this->autoMapping->mapToObject(StoreOwnerBranchDeleteRequest::class, StoreOwnerBranchEntity::class, $request, $entity);
+
+            $this->entityManager->flush();
+
+            return $entity;
+        }
+
+        return StoreOwnerBranch::BRANCH_NOT_FOUND;
+    }
+
+    /**
+     * @param $storeOwnerId
+     * @return array
+     */
+    public function getAllBranches($storeOwnerId): array
+    {
+       return $this->storeOwnerBranchEntityRepository->findBy(['storeOwner' => $storeOwnerId]);
```

Comment:
Why we don't use a custom function in the repository layer to retrieve all branches of provided store owner?

Reply:
@ralsalloum it is totally fine to use findBy();